fe/fe-core/src/main/java/org/apache/doris/master/ReportHandler.java

```diff
@@ -962,10 +963,19 @@ private static void deleteFromMeta(ListMultimap<Long, Long> tabletDeleteFromMeta
                                             objectPool,
                                             olapTable.rowStorePageSize(),
                                             olapTable.variantEnableFlattenNested());
-
                                     createReplicaTask.setIsRecoverTask(true);
                                     createReplicaTask.setInvertedIndexFileStorageFormat(olapTable
                                                                 .getInvertedIndexFileStorageFormat());
+                                    if (indexId == olapTable.getBaseIndexId() || olapTable.isShadowIndex(indexId)) {
+                                        List<Integer> clusterKeyIndexes = OlapTable.getClusterKeyIndexes(
+                                                indexMeta.getSchema());
+                                        if (!CollectionUtils.isEmpty(clusterKeyIndexes)) {
+                                            createReplicaTask.setClusterKeyIndexes(clusterKeyIndexes);
+                                            LOG.info("table: {}, partition: {}, index: {}, tablet: {}, "
+                                                            + "cluster key indexes: {}", tableId, partitionId, indexId,
+                                                    tabletId, clusterKeyIndexes);
+                                        }
+                                    }
```

Comment:
May be we should pass olaptable as arg to createReplicaTask and handle clusterkeyindex in createReplicaTask internally.